tests/e2e/test_rosacli_network_resources.go

```diff
@@ -0,0 +1,301 @@
+				output, err = networkResourcesService.CreateNetworkResources(false, "single-vpc",
+					paramNameFlag,
+					paramRegionFlag,
+					"--template-dir", templateDirPath,
+					"--param=AvailabilityZoneCount=3",
+					"--param=Tags=Key1=Value1,Key2=Value2",
+					"--mode", "manual")
+				defer func() {
+					params := cloudformation.DeleteStackInput{
+						StackName: &stackName_2,
+					}
+					_, err = awsClient.StackFormationClient.DeleteStack(context.TODO(), &params)
+					Expect(err).ToNot(HaveOccurred())
+				}()
+				Expect(err).ToNot(HaveOccurred())
+				resp = rosaClient.Parser.TextData.Input(output).Parse().Output()
+				Expect(resp).To(
+					ContainSubstring(manualModeOutput))
+
+				By("Try to create network by setting OCM_TEMPLATE_DIR env variable")
+				stackName_3 := helper.GenerateRandomName("ocp-77140", 2)
+				paramNameFlag = fmt.Sprintf("--param=Name=%s", stackName_3)
+				output, err = networkResourcesService.CreateNetworkResources(true, "single-vpc",
+					paramNameFlag,
+					paramRegionFlag)
+				defer func() {
+					params := cloudformation.DeleteStackInput{
+						StackName: &stackName_3,
+					}
+					_, err = awsClient.StackFormationClient.DeleteStack(context.TODO(), &params)
+					Expect(err).ToNot(HaveOccurred())
+				}()
+				Expect(err).ToNot(HaveOccurred())
+				resp = rosaClient.Parser.TextData.Input(output).Parse().Output()
+				Expect(resp).To(
+					ContainSubstring("msg=\"Stack %s created\"", stackName_3))
+
+				By("Try to override 'OCM_TEMPLATE_DIR' env variable using --template-dir flag")
+				stackName_4 := helper.GenerateRandomName("ocp-77140", 2)
+				paramNameFlag = fmt.Sprintf("--param=Name=%s", stackName_4)
+				output, err = networkResourcesService.CreateNetworkResources(true, "single-vpc",
+					paramNameFlag,
+					paramRegionFlag,
+					"--template-dir", templateDirPath)
+				defer func() {
+					params := cloudformation.DeleteStackInput{
+						StackName: &stackName_4,
+					}
+					_, err = awsClient.StackFormationClient.DeleteStack(context.TODO(), &params)
+					Expect(err).ToNot(HaveOccurred())
+				}()
+				Expect(err).ToNot(HaveOccurred())
+				resp = rosaClient.Parser.TextData.Input(output).Parse().Output()
+				Expect(resp).To(
+					ContainSubstring("msg=\"Stack %s created\"", stackName_4))
+			})
+
+		It("should be validated successfully - [id:77159]",
+			labels.Medium, labels.Runtime.OCMResources,
+			func() {
+				By("Create aws client")
+				awsClient, err := aws_client.CreateAWSClient("", "us-west-2")
+				Expect(err).ToNot(HaveOccurred())
+
+				By("Create template dir for template file missing Region Param")
+				templateContent := helper.TemplateWithoutRegionParam()
+				templatePath_1, err := helper.CreateTemplateDirForNetworkResources("without-region", templateContent)
+				defer func() {
+					os.Remove(templatePath_1)
+					time.Sleep(time.Second * 5) // wait until template file is deleted
+					os.Remove("without-region")
+				}()
+				Expect(err).ToNot(HaveOccurred())
```

Comment:
We need to check the output and the stack created as expected. In this step By("Create template dir for template file missing Region Param"), if should use the default region, there will be INFO message like "Region not provided, using default region us-east-2". And test case should have checkpoint, at least in one place, to check if the stack uses the default region.

Same as other checkpoints bellow which uses the default value when the command is set related flags.

Reply:
No here, region parameter is not set in the template file itself, then it will return error

Reply:
For default value I have added in the test case 77140 at the first step